Deleting App Sign-on Policy Fails in 4.13.0 and 4.13.1

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

```
Terraform v1.10.5
on darwin_arm64
+ provider registry.terraform.io/okta/okta v4.13.1
```

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

- okta_app_signon_policy

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
terraform {
  required_providers {
    okta = {
      source  = "okta/okta"
      version = "4.13.1"
    }
  }
}

# provider "okta" {
#   org_name  = "<REDACTED>"
#   base_url  = "okta.com"
#   api_token = "<REDACTED>"
# }

# Memo: comment out the following block after the policy is created
resource "okta_app_signon_policy" "policy" {
  name        = "test_policy"
  description = "Test policy"
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

```hcl
okta_app_signon_policy.policy: Refreshing state... [id=<REDACTED_POLICY_ID>]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # okta_app_signon_policy.policy will be destroyed
  # (because okta_app_signon_policy.policy is not in configuration)
  - resource "okta_app_signon_policy" "policy" {
      - catch_all   = true -> null
      - description = "Test policy" -> null
      - id          = "<REDACTED_POLICY_ID>" -> null
      - name        = "test_policy" -> null
    }

Plan: 0 to add, 0 to change, 1 to destroy.
```

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

```
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-okta_v4.13.1 plugin:

panic: interface conversion: interface is nil, not okta.OktaApp

goroutine 61 [running]:
github.com/okta/terraform-provider-okta/okta.(*appSignOnPolicyResource).Delete(0x14000786030, {0x10668ac58, 0x1400034f650}, {{{{0x106690220, 0x1400074c660}, {0x106332ba0, 0x1400074c4b0}}, {0x106692f50, 0x140007a1590}}, {{{0x0, ...}, ...}, ...}, ...}, ...)
        github.com/okta/terraform-provider-okta/okta/resource_okta_app_signon_policy.go:234 +0x41c
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).DeleteResource(0x140002791e0, {0x10668ac58, 0x1400034f650}, 0x140004df408, 0x140007633b8)
        github.com/hashicorp/terraform-plugin-framework@v1.8.0/internal/fwserver/server_deleteresource.go:100 +0x4ec
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0x140004df5a0?, {0x10668ac58, 0x1400034f650}, 0x1400027ebe0, 0x140007635a0)
        github.com/hashicorp/terraform-plugin-framework@v1.8.0/internal/fwserver/server_applyresourcechange.go:79 +0x248
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0x140002791e0, {0x10668ac58?, 0x1400034f470?}, 0x1400027eaa0)
        github.com/hashicorp/terraform-plugin-framework@v1.8.0/internal/proto6server/server_applyresourcechange.go:55 +0x314
github.com/hashicorp/terraform-plugin-mux/tf6to5server.v6tov5Server.ApplyResourceChange({{0x106693938?, 0x140002791e0?}}, {0x10668ac58, 0x1400034f470}, 0x0?)
        github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf6to5server/tf6to5server.go:59 +0x58
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0x10668ac90?, {0x10668ac58?, 0x1400034edb0?}, 0x1400027ea50)
        github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x184
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x140002a8e60, {0x10668ac58?, 0x1400034e2a0?}, 0x14000232000)
        github.com/hashicorp/terraform-plugin-go@v0.22.2/tfprotov5/tf5server/server.go:846 +0x2b0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1065bcc40?, 0x140002a8e60}, {0x10668ac58, 0x1400034e2a0}, 0x14000028200, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.22.2/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001b5200, {0x10668ac58, 0x14000570300}, {0x106691498, 0x140001c6180}, 0x1400056c900, 0x1400048ba70, 0x107566938, 0x0)
        google.golang.org/grpc@v1.63.2/server.go:1369 +0xba0
google.golang.org/grpc.(*Server).handleStream(0x140001b5200, {0x106691498, 0x140001c6180}, 0x1400056c900)
        google.golang.org/grpc@v1.63.2/server.go:1780 +0xc80
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.63.2/server.go:1019 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 15
        google.golang.org/grpc@v1.63.2/server.go:1030 +0x150

Error: The terraform-provider-okta_v4.13.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

<!--- What should have happened? --->
The policy should be deleted successfully.

#### Can this be done in the Admin UI?

<!--- Yes / No / I don't know --->
Yes: policy deletion is possible

#### Can this be done in the actual API call?

<!--- Yes / No / I don't know --->
Yes: policy deletion through console is still possible, and deletion in 4.12.0 still works

### Actual Behavior

<!--- What actually happened? --->

The Terraform provider panics due to a cast from `nil` to `okta.OktaApp` when enumerating apps with the given app sign-on policy.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply` the given Terraform code. (Use provider version 4.13.1)
2. Comment out the `okta_app_signon_policy`.
3. `terraform apply` again.
4. See that the apply fails and the provider crashes.

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->
N/A

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

- This is the same problem as the later half as https://github.com/okta/terraform-provider-okta/issues/2182, which is closed.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Changelog
 
+## 4.14.1 (Mar 03, 2025)
+
+### BUG FIXES
+
+* Fix `default_rule_id` argument error in `okta_app_signon_policy` introduced in v4.13.0 / v4.13.1 [#2240](https://github.com/okta/terraform-provider-okta/pull/2240). Thanks, [@monde](https://github.com/monde)!
+
 ## 4.14.0 (Feb 11, 2025)
 
 ### IMPROVEMENTS

--- a/docs/index.md
+++ b/docs/index.md
@@ -24,7 +24,7 @@ terraform {
   required_providers {
     okta = {
       source = "okta/okta"
-      version = "~> 4.14.0"
+      version = "~> 4.14.1"
     }
   }
 }

--- a/docs/resources/app_signon_policy.md
+++ b/docs/resources/app_signon_policy.md
@@ -98,7 +98,7 @@ resource "okta_app_signon_policy_rule" "some_rule" {
 
 ### Read-Only
 
-- `default_rule_id` (String) Default rules id of the policy
+- `default_rule_id` (String) Default rule (system=true) id of the policy
 - `id` (String) Policy id
 
 ## Import

--- a/docs/resources/app_signon_policy.md
+++ b/docs/resources/app_signon_policy.md
@@ -94,7 +94,7 @@ resource "okta_app_signon_policy_rule" "some_rule" {
 
 ### Optional
 
-- `catch_all` (Boolean) Default rules of the policy set to `DENY` or not. If `false`, it is set to `DENY`. **WARNING** setting this attribute to false change the OKTA default behavior. Use at your own risk. This is only apply during creation, so import or update will not work
+- `catch_all` (Boolean, default true, creation-only argument) If false, the default rule of the policy is set access to `DENY`. Otherwise default behavior of the default rule is to leave access at `ALLOW`.  **WARNING** setting this attribute to false changes policy rule's default behavior. Use at your own risk. This is only applied during creation and does not affect import or update.
 
 ### Read-Only
 

--- a/okta/config.go
+++ b/okta/config.go
@@ -26,7 +26,7 @@ import (
 )
 
 const (
-	OktaTerraformProviderVersion   = "4.14.0"
+	OktaTerraformProviderVersion   = "4.14.1"
 	OktaTerraformProviderUserAgent = "okta-terraform/" + OktaTerraformProviderVersion
 )
 

--- a/okta/resource_okta_app_signon_policy.go
+++ b/okta/resource_okta_app_signon_policy.go
@@ -2,6 +2,7 @@ package okta
 
 import (
 	"context"
+	"errors"
 	"path"
 
 	"github.com/hashicorp/terraform-plugin-framework/diag"
@@ -79,7 +80,7 @@ default/system access policy.`,
 				Default:     booldefault.StaticBool(true),
 			},
 			"default_rule_id": schema.StringAttribute{
-				Description: "Default rules id of the policy",
+				Description: "Default rule (system=true) id of the policy",
 				Computed:    true,
 				PlanModifiers: []planmodifier.String{
 					stringplanmodifier.UseStateForUnknown(),
@@ -113,43 +114,28 @@ func (r *appSignOnPolicyResource) Create(ctx context.Context, req resource.Creat
 		return
 	}
 
-	resp.Diagnostics.Append(mapAccessPolicyToState(accessPolicy, &state)...)
+	resp.Diagnostics.Append(r.mapAccessPolicyToState(ctx, accessPolicy, &state)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	rules, _, err := r.oktaSDKClientV5.PolicyAPI.ListPolicyRules(ctx, accessPolicy.AccessPolicy.GetId()).Execute()
-	if err != nil {
-		resp.Diagnostics.AddError(
-			"failed to get default access policy rule",
-			err.Error(),
-		)
-		return
-	}
-	if len(rules) != 1 {
-		resp.Diagnostics.AddError(
-			"find more than one default access policy rule",
-			"",
-		)
-		return
-	}
-	if rules[0].AccessPolicyRule == nil {
-		resp.Diagnostics.AddError(
-			"failed to find default access policy rule",
-			"",
-		)
-		return
-	}
-	defaultRuleID := rules[0].AccessPolicyRule.GetId()
-	state.DefaultRuleID = types.StringValue(defaultRuleID)
-
 	if !state.CatchAll.ValueBool() {
-		if actions, ok := rules[0].AccessPolicyRule.GetActionsOk(); ok {
+		defaultRule, err := r.findDefaultPolicyRuleResponse(ctx, state.ID.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"failed to find default policy rule",
+				err.Error(),
+			)
+			return
+		}
+		if actions, ok := defaultRule.AccessPolicyRule.GetActionsOk(); ok {
 			if _, ok := actions.GetAppSignOnOk(); ok {
-				rules[0].AccessPolicyRule.Actions.AppSignOn.SetAccess("DENY")
+				defaultRule.AccessPolicyRule.Actions.AppSignOn.SetAccess("DENY")
 			}
 		}
-		_, _, err = r.oktaSDKClientV5.PolicyAPI.ReplacePolicyRule(ctx, accessPolicy.AccessPolicy.GetId(), defaultRuleID).PolicyRule(rules[0]).Execute()
+		defaultRule.AccessPolicyRule.Actions.AppSignOn.SetAccess("DENY")
+
+		_, _, err = r.oktaSDKClientV5.PolicyAPI.ReplacePolicyRule(ctx, state.ID.ValueString(), state.DefaultRuleID.ValueString()).PolicyRule(*defaultRule).Execute()
 		if err != nil {
 			resp.Diagnostics.AddError(
 				"failed to update access policy default rule to DENY",
@@ -185,7 +171,7 @@ func (r *appSignOnPolicyResource) Read(ctx context.Context, req resource.ReadReq
 		return
 	}
 
-	resp.Diagnostics.Append(mapAccessPolicyToState(accessPolicy, &state)...)
+	resp.Diagnostics.Append(r.mapAccessPolicyToState(ctx, accessPolicy, &state)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
@@ -281,7 +267,7 @@ func (r *appSignOnPolicyResource) Update(ctx context.Context, req resource.Updat
 		return
 	}
 
-	resp.Diagnostics.Append(mapAccessPolicyToState(accessPolicy, &state)...)
+	resp.Diagnostics.Append(r.mapAccessPolicyToState(ctx, accessPolicy, &state)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
@@ -304,7 +290,7 @@ func buildV5AccessPolicy(model appSignOnPolicyResourceModel) okta.ListPolicies20
 	return okta.ListPolicies200ResponseInner{AccessPolicy: accessPolicy}
 }
 
-func mapAccessPolicyToState(data *okta.ListPolicies200ResponseInner, state *appSignOnPolicyResourceModel) diag.Diagnostics {
+func (r *appSignOnPolicyResource) mapAccessPolicyToState(ctx context.Context, data *okta.ListPolicies200ResponseInner, state *appSignOnPolicyResourceModel) diag.Diagnostics {
 	var diags diag.Diagnostics
 	if data.AccessPolicy == nil {
 		diags.AddError("Empty response", "Access policy")
@@ -313,7 +299,37 @@ func mapAccessPolicyToState(data *okta.ListPolicies200ResponseInner, state *appS
 	state.ID = types.StringPointerValue(data.AccessPolicy.Id)
 	state.Name = types.StringPointerValue(data.AccessPolicy.Name)
 	state.Description = types.StringPointerValue(data.AccessPolicy.Description)
+
+	defaultRule, err := r.findDefaultPolicyRuleResponse(ctx, state.ID.ValueString())
+	if err != nil {
+		diags.AddError(
+			"failed to get default access policy rule",
+			err.Error(),
+		)
+		return diags
+	}
+	defaultRuleID := defaultRule.AccessPolicyRule.GetId()
+	state.DefaultRuleID = types.StringValue(defaultRuleID)
+
 	return diags
 }
 
-// TODU double check crud and rerun the test
+// findDefaultPolicyRuleResponse find the default policy rule from the list and return
+// it. Default rule is the first to be marked system.
+func (r *appSignOnPolicyResource) findDefaultPolicyRuleResponse(ctx context.Context, accessPolicyId string) (*okta.ListPolicyRules200ResponseInner, error) {
+	rules, _, err := r.oktaSDKClientV5.PolicyAPI.ListPolicyRules(ctx, accessPolicyId).Execute()
+	if err != nil {
+		return nil, err
+	}
+	for _, rule := range rules {
+		if rule.AccessPolicyRule == nil {
+			continue
+		}
+		if system, ok := rule.AccessPolicyRule.GetSystemOk(); ok {
+			if *system {
+				return &rule, nil
+			}
+		}
+	}
+	return nil, errors.New("policy does not have a default (system) access policy rule")
+}

--- a/okta/resource_okta_app_signon_policy.go
+++ b/okta/resource_okta_app_signon_policy.go
@@ -231,6 +231,10 @@ func (r *appSignOnPolicyResource) Delete(ctx context.Context, req resource.Delet
 	}
 
 	for _, a := range apps {
+		if a.GetActualInstance() == nil {
+			// nil bumper
+			continue
+		}
 		app := a.GetActualInstance().(OktaApp)
 		accessPolicy := app.GetLinks().AccessPolicy.GetHref()
 		// ignore apps that don't have an access policy, typically Classic org apps.

--- a/okta/resource_okta_app_signon_policy.go
+++ b/okta/resource_okta_app_signon_policy.go
@@ -74,7 +74,7 @@ default/system access policy.`,
 				Required:    true,
 			},
 			"catch_all": schema.BoolAttribute{
-				Description: "Default rules of the policy set to `DENY` or not. If `false`, it is set to `DENY`. **WARNING** setting this attribute to false change the OKTA default behavior. Use at your own risk. This is only apply during creation, so import or update will not work",
+				Description: "If false, the default rule of the policy is set access to `DENY`. Otherwise default behavior of the default rule is to leave access at `ALLOW`.  **WARNING** setting this attribute to false changes policy rule's default behavior. Use at your own risk. This is only applied during creation and does not affect import or update.",
 				Optional:    true,
 				Computed:    true,
 				Default:     booldefault.StaticBool(true),

--- a/okta/resource_okta_app_signon_policy_test.go
+++ b/okta/resource_okta_app_signon_policy_test.go
@@ -1,10 +1,12 @@
 package okta
 
 import (
+	"errors"
 	"fmt"
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
 func TestAccResourceOktaAppSignOnPolicy_crud(t *testing.T) {
@@ -13,6 +15,11 @@ func TestAccResourceOktaAppSignOnPolicy_crud(t *testing.T) {
 	updatedConfig := mgr.GetFixtures("basic_updated.tf", t)
 	renamedConfig := mgr.GetFixtures("basic_renamed.tf", t)
 	resourceName := fmt.Sprintf("%v.test", appSignOnPolicy)
+
+	importConfig := mgr.ConfigReplace(`
+resource "okta_app_signon_policy_rule" "test" {
+  depends_on = [okta_app_signon_policy.test]
+}`)
 
 	oktaResourceTest(t, resource.TestCase{
 		PreCheck:                 testAccPreCheck(t),
@@ -26,6 +33,7 @@ func TestAccResourceOktaAppSignOnPolicy_crud(t *testing.T) {
 					ensurePolicyExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "name", buildResourceNameWithPrefix("testAcc_Test_App", mgr.Seed)),
 					resource.TestCheckResourceAttr(resourceName, "description", "The app signon policy used by our test app."),
+					resource.TestCheckResourceAttrSet(resourceName, "default_rule_id"),
 				),
 			},
 			{
@@ -34,6 +42,7 @@ func TestAccResourceOktaAppSignOnPolicy_crud(t *testing.T) {
 					ensurePolicyExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "name", buildResourceNameWithPrefix("testAcc_Test_App", mgr.Seed)),
 					resource.TestCheckResourceAttr(resourceName, "description", "The updated app signon policy used by our test app."),
+					resource.TestCheckResourceAttrSet(resourceName, "default_rule_id"),
 				),
 			},
 			{
@@ -42,7 +51,109 @@ func TestAccResourceOktaAppSignOnPolicy_crud(t *testing.T) {
 					ensurePolicyExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "name", buildResourceNameWithPrefix("testAcc_Test_App_Renamed", mgr.Seed)),
 					resource.TestCheckResourceAttr(resourceName, "description", "The app signon policy used by our test app."),
+					resource.TestCheckResourceAttrSet(resourceName, "default_rule_id"),
 				),
+			},
+			// check the default_rule_id was set by looking up the real policy rule by id and check its access is set to ALLOW
+			{
+				Config:       fmt.Sprintf("%s\n%s", renamedConfig, importConfig),
+				ResourceName: "okta_app_signon_policy_rule.test",
+				ImportState:  true,
+				ImportStateIdFunc: func(s *terraform.State) (string, error) {
+					rs, ok := s.RootModule().Resources["okta_app_signon_policy.test"]
+					if !ok {
+						return "", fmt.Errorf("failed to find policy")
+					}
+
+					policyId := rs.Primary.Attributes["id"]
+					defaultRuleId := rs.Primary.Attributes["default_rule_id"]
+
+					return fmt.Sprintf("%s/%s", policyId, defaultRuleId), nil
+				},
+				ImportStateCheck: func(states []*terraform.InstanceState) error {
+					if len(states) != 1 {
+						return errors.New("failed to import schema into state")
+					}
+					instance := states[0]
+					access := instance.Attributes["access"]
+					if access != "ALLOW" {
+						return fmt.Errorf("expected okta_app_signon_policy.test access to be set to ALLOW, got %s", access)
+					}
+
+					return nil
+				},
+			},
+		},
+	})
+}
+
+// TestAccResourceOktaAppSignOnPolicy_default_policy_rule_can_be_set_to_deny
+// tests to see if catch_all = false results in a default policy rule of DENY
+func TestAccResourceOktaAppSignOnPolicy_default_policy_rule_can_be_set_to_deny(t *testing.T) {
+	resourceName := fmt.Sprintf("%v.test", appSignOnPolicy)
+	mgr := newFixtureManager("resources", appSignOnPolicy, t.Name())
+	config := `
+resource "okta_app_signon_policy" "test" {
+  name        = "testAcc_Test_App_replace_with_uuid"
+  description = "App Sign-On Policy with Default Rule DENY"
+  catch_all   = false
+}`
+	importConfig := `
+resource "okta_app_signon_policy" "test" {
+  name        = "testAcc_Test_App_replace_with_uuid"
+  description = "App Sign-On Policy with Default Rule DENY"
+  catch_all   = false
+}
+resource "okta_app_signon_policy_rule" "test" {
+  depends_on = [okta_app_signon_policy.test]
+}`
+	oktaResourceTest(t, resource.TestCase{
+		PreCheck:                 testAccPreCheck(t),
+		ErrorCheck:               testAccErrorChecks(t),
+		ProtoV5ProviderFactories: testAccMergeProvidersFactories,
+		CheckDestroy:             checkPolicyDestroy(appSignOnPolicy),
+		Steps: []resource.TestStep{
+			{
+				Config: mgr.ConfigReplace(config),
+				Check: resource.ComposeTestCheckFunc(
+					ensurePolicyExists(resourceName),
+					resource.TestCheckResourceAttr("okta_app_signon_policy.test", "name", buildResourceNameWithPrefix("testAcc_Test_App", mgr.Seed)),
+					resource.TestCheckResourceAttr("okta_app_signon_policy.test", "description", "App Sign-On Policy with Default Rule DENY"),
+					resource.TestCheckResourceAttr("okta_app_signon_policy.test", "catch_all", "false"),
+					resource.TestCheckResourceAttrSet("okta_app_signon_policy.test", "default_rule_id"),
+				),
+			},
+			{
+				Config:       mgr.ConfigReplace(importConfig),
+				ResourceName: "okta_app_signon_policy_rule.test",
+				ImportState:  true,
+
+				// fyi, import state id func equivelent to
+				// terraform import okta_app_signon_policy_rule.test [policy id]/[policy rule id]
+				// and is dirived directly off the attributes on okta_app_signon_policy.test
+				ImportStateIdFunc: func(s *terraform.State) (string, error) {
+					rs, ok := s.RootModule().Resources["okta_app_signon_policy.test"]
+					if !ok {
+						return "", fmt.Errorf("failed to find policy")
+					}
+
+					policyId := rs.Primary.Attributes["id"]
+					defaultRuleId := rs.Primary.Attributes["default_rule_id"]
+
+					return fmt.Sprintf("%s/%s", policyId, defaultRuleId), nil
+				},
+				ImportStateCheck: func(states []*terraform.InstanceState) error {
+					if len(states) != 1 {
+						return errors.New("failed to import schema into state")
+					}
+					instance := states[0]
+					access := instance.Attributes["access"]
+					if access != "DENY" {
+						return fmt.Errorf("expected okta_app_signon_policy.test access to be set to DENY, got %s", access)
+					}
+
+					return nil
+				},
 			},
 		},
 	})

--- a/templates/index.md
+++ b/templates/index.md
@@ -24,7 +24,7 @@ terraform {
   required_providers {
     okta = {
       source = "okta/okta"
-      version = "~> 4.14.0"
+      version = "~> 4.14.1"
     }
   }
 }

